cs3api4lab/auth/eos.py

```diff
@@ -0,0 +1,38 @@
+import os
+from io import TextIOWrapper
+
+from jupyter_core.paths import jupyter_config_path
+from notebook.services.config import ConfigManager
+
+from cs3api4lab.auth.authenticator import Authenticator
+
+
+class Eos(Authenticator):
+    tokens = {}
+
+    def __init__(self, config=None, log=None):
+
+        self.config = config
+        file = None
+        if "eos_file" in self.config and self.config['eos_file'] is not "":
+
+            try:
```

Comment:
https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files

> It is good practice to use the with keyword when dealing with file objects. The advantage is that the file is properly closed after its suite finishes, even if an exception is raised at some point. Using with is also much shorter than writing equivalent try-finally blocks:

Reply:
Do not resolve issues without a fix.

Reply:
It is fixed in the last commit. Line 25 and 26 in cs3api4lab/auth/eos.py

Reply:
True. My bad!